src/yio-model/mediaplayer/tvchannelmodel_mediaplayer.h

```diff
@@ -0,0 +1,135 @@
+ *****************************************************************************/
+
+#pragma once
+
+#include <QAbstractListModel>
+#include <QObject>
+#include <QVariant>
+#include <QtDebug>
+
+class tvchannelModelItem {
+ public:
+    tvchannelModelItem(const QString& key, const QString& time, const QString& title,
+                       const QString& subtitle, const QString& type,
+              const QString& imageUrl, const QVariant& commands)
+        : m_key(key), m_time(time), m_title(title), m_subtitle(subtitle),
+          m_type(type), m_imageUrl(imageUrl), m_commands(commands) {}
+
+    QString  item_key() const { return m_key; }
+    QString  item_time() const { return m_time; }
+    QString  item_title() const { return m_title; }
+    QString  item_subtitle() const { return m_subtitle; }
+    QString  item_type() const { return m_type; }
+    QString  item_imageUrl() const { return m_imageUrl; }
+    QVariant item_commands() const { return m_commands; }
+
+ private:
+    QString  m_key;
+    QString  m_time;
+    QString  m_title;
+    QString  m_subtitle;
+    QString  m_type;
+    QString  m_imageUrl;
+    QVariant m_commands;
+};
+
+class ListtvchannelModel : public QAbstractListModel {
+    Q_OBJECT
+    Q_PROPERTY(int count READ count WRITE setCount NOTIFY countChanged)
+
+ public:
+    enum SearchRoles { KeyRole = Qt::UserRole + 1, TimeRole, TitleRole, SubTitleRole,
+                       TypeRole, ImageUrlRole, CommandsRole };
+
+    explicit ListtvchannelModel(QObject* parent = nullptr);
+    ~ListtvchannelModel() {}
+
+    int                    count() const;
+    int                    rowCount(const QModelIndex& p = QModelIndex()) const;
+    QVariant               data(const QModelIndex& index, int role = Qt::DisplayRole) const;
+    QHash<int, QByteArray> roleNames() const;
+
+    void append(const tvchannelModelItem& o);
+
+ public slots:  // NOLINT open issue: https://github.com/cpplint/cpplint/pull/99
+    void setCount(int count);
+
+ signals:
+    void countChanged(int count);
+
+ private:
+    int              m_count;
+    QList<tvchannelModelItem> m_data;
+};
+
+class BrowsetvchannelModel : public QObject {
+    Q_OBJECT
+    Q_PROPERTY(QString id READ id NOTIFY idChanged)
+    Q_PROPERTY(QString title READ title NOTIFY titleChanged)
+    Q_PROPERTY(QString subtitle READ subtitle NOTIFY subtitleChanged)
+    Q_PROPERTY(QString type READ type NOTIFY typeChanged)
+    Q_PROPERTY(QString imageUrl READ imageUrl NOTIFY imageUrlChanged)
+    Q_PROPERTY(QObject* model READ model NOTIFY modelChanged)
+    Q_PROPERTY(QStringList commands READ commands NOTIFY commandsChanged)
+
+ public:
+    BrowsetvchannelModel(QObject* parent = nullptr, const QString& id = "", const QString& time = "",
+                         const QString& title = "", const QString& subtitle = "", const QString& type = "",
+                         const QString& imageUrl = "", const QStringList& commands = {})
```

Comment:
QString initialization to empty string with `""` is usually not required. They are initialized with a null-string by default, which will be the same as an empty string for most operations. One of the few differences is when using `isNull()`:
https://doc.qt.io/qt-5/qstring.html#distinction-between-null-and-empty-strings

Reply:
This is needed due to the fact QCreator is requiring it:

`tvchannelmodel_mediaplayer.h:96:68: error: missing default argument on parameter `

Reply:
This is not needed for Qt Creator, but how the client instantiates the class.
The `QObject* parent = nullptr` argument should always be the last argument. In Qt the parent argument is optional and only used when child objects are allocated and automatically freed when the parent is freed.

Unfortunately this is also wrong in the existing `BrowseModel`. 
Furthermore, the parent parameter needs to be delegated to the base class.

Reply:
solved